spec/requests/api/v0/highlights_controller_spec.rb

```diff
@@ -147,18 +127,46 @@
       before(:each) { stub_current_user_uuid(user_uuid) }
 
       context 'when the request is valid' do
-        before { post highlights_path, params: valid_attributes }
-
-        it 'creates a highlight' do
-          expect(json_response[:anchor]).to eq('foo anchor')
+        before do
+          post highlights_path, params: valid_attributes
+          @hl1_id = json_response[:id]
         end
 
-        it 'returns status code 201' do
-          expect(response).to have_http_status(201)
+        context 'when there are no highlights yet' do
+          it 'creates a highlight' do
+            expect(json_response[:anchor]).to eq('foo anchor')
+          end
+
+          it 'returns status code 201' do
+            expect(response).to have_http_status(201)
+          end
+
+          it 'is owned by the right user' do
+            expect(Highlight.first.user_uuid).to eq user_uuid
+          end
         end
 
-        it 'is owned by the right user' do
-          expect(Highlight.first.user_uuid).to eq user_uuid
+        context 'when there is one highlight' do
+          # Lots more invalid data tests in the model spec
+          it '422s when the new highlight does not specify prev or next' do
+            post highlights_path, params: valid_attributes
+            expect(response).to have_http_status(:unprocessable_entity)
+            expect(response.body).to match(/Must specify previous or next highlight/)
+          end
+
+          it 'puts the new highlight after the first one when set a prev highlight' do
```

Comment:
does this need to explicitly test hl1 and new HLs' prev and next ids are all correct?

Reply:
good point, I added one case of this in this spec and then more in the model spec where they were missing.